pkg/reconciler/trigger/trigger.go

```diff
@@ -106,7 +106,13 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, t *brokerv1beta1.Trigger
 		// Call Finalizer anyway in case the Trigger still holds GCP Broker related resources.
 		// If a Trigger used to point to a GCP Broker but now has a Broker with a different brokerclass,
 		// we should clean up resources related to GCP Broker.
-		return r.FinalizeKind(ctx, t)
+		event := r.FinalizeKind(ctx, t)
+		// It impossible to tell if the trigger has ever pointed to a GCP broker. Mark topic ready to
+		// unblock the user of other brokers. This should be deleted once the TODO is finished
```

Comment:
"It's impossible".

It would be great to add an example here. Like previously was referencing a GCP broker, then it's deleted and recreated as non-gcp broker. We still want to best effort GC the resources but don't want to block the trigger being ready with the new broker.

Reply:
Added more detailed comments which the change deserves. Thanks for the suggestion!